crates/node/src/validate.rs

```diff
@@ -414,9 +417,9 @@ impl StateRead for State {
                 }
                 Ok(values)
             })
-            .await?
-        }
-        .boxed()
+            .await
+            .map_err(StateReadError::from)
+        })
     }
 }
```

Comment:
The `conn_pool` used here is actually a special `Db` type introduced to support both dry runs and regular validation - I think we can just leave this as it previously was

Reply:
I'm not sure here if you mean just these lines, or to revert the full changes I made to the `key_range()` function.
Happy to roll back if you think that's the best way to proceed here. :)